pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -2274,6 +2289,21 @@ private void updateManagedLedgerConfig() {
         configRegisteredListeners.put(configKey, listener);
     }
 
+    public <T> void registerPublishBufferLimitListener(Consumer<PublishBufferEvent> listener) {
+        publishBufferLimitListeners.add(listener);
+    }
+
+    private void checkPulishBufferLimit() {
+        publishBufferLimitListeners.forEach(listener -> {
+            if (maxPendingBytes > 0 && totalPendingBytes.sum() > maxPendingBytes) {
```

Comment:
We do not need to call sum() for each listener in this method. We could use follow flow:
Step1: check if maxPendingBytes > 0
Step2: do sum() and compare with threshold
Step3: inform each listener if necessary

Reply:
@wangjialing218  PTAL again

Reply:
LGTM